src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -336,6 +344,40 @@ void IdealGraphPrinter::begin_method() {
 
   print_prop(COMPILATION_ID_PROPERTY, C->compile_id());
 
+  stringStream args;
+  Arguments::print_jvm_args_on(&args);
+  print_prop(COMPILATION_ARGUMENTS_PROPERTY, args.freeze());
+
+  stringStream machine;
+  buffer[0] = 0;
+  os::print_summary_info(&machine, buffer, sizeof(buffer) - 1);
+  print_prop(COMPILATION_MACHINE_PROPERTY, machine.freeze());
+
+  stringStream version;
+  buffer[0] = 0;
+  JDK_Version::current().to_string(buffer, sizeof(buffer) - 1);
+  const char* runtime_name = JDK_Version::runtime_name() != nullptr ?
+    JDK_Version::runtime_name() : "";
+  const char* runtime_version = JDK_Version::runtime_version() != nullptr ?
+    JDK_Version::runtime_version() : "";
+  const char* vendor_version = JDK_Version::runtime_vendor_version() != nullptr ?
+    JDK_Version::runtime_vendor_version() : "";
+  const char* jdk_debug_level = VM_Version::printable_jdk_debug_level() != nullptr ?
+    VM_Version::printable_jdk_debug_level() : "";
+
+  version.print_cr("%s%s%s (%s) (%sbuild %s)", runtime_name,
+                   (*vendor_version != '\0') ? " " : "", vendor_version,
+                   buffer, jdk_debug_level, runtime_version);
+  print_prop(COMPILATION_VM_VERSION_PROPERTY, version.freeze());
+
+  stringStream time;
+  buffer[0] = 0;
+  os::print_date_and_time(&time, buffer, sizeof(buffer) - 1);
+  print_prop(COMPILATION_DATE_TIME_PROPERTY, time.freeze());
+
+  print_prop(COMPILATION_PROCESS_ID_PROPERTY, os::current_process_id());
+  print_prop(COMPILATION_THREAD_ID_PROPERTY, os::current_thread_id());
+
```

Comment:
What about CPU features? Could be nice to know if we have `avx2` or `asimd`, etc.

Reply:
Done (commit cb781a95).